File: Problems/Solved/Easy/1260.js
/**
 * @param {number[][]} grid
 * @param {number} k
 * @return {number[][]}
 */
var shiftGrid = function (grid, k) {
    if (grid.length === 1 && grid[0].length === 1) return grid;
    let cache = [];
    for (let i = 0; i < grid.length; i++) {
        cache.push(...grid[i]);
    }
    if (k > cache.length) {
        k = k % cache.length;
    }
    let numsShift = cache.slice(cache.length - k, cache.length);
    numsShift.push(...cache);
    let result = [];
    console.log(numsShift);
    console.log(grid[0].length);
    for (let j = 0; j < numsShift.length - k; j += 0) {
        result.push(numsShift.slice(j, j + grid[0].length));
        j += grid[0].length;
    }
    return result;
};

grid = [[1], [2], [3], [4], [7], [6], [5]];
k = 23;

console.log(shiftGrid(grid, k));

File: Problems/Solved/Easy/1260.rs
pub fn shift_grid(grid: Vec<Vec<i32>>, k: i32) -> Vec<Vec<i32>> {
    if grid.len() == 1 && grid[0].len() == 1 {
        return grid;
    }
    let mut cache: Vec<i32> = vec![];
    for row in &grid {
        cache.extend_from_slice(row);
    }
    let cache_len = cache.len();
    let mut k = k as usize;
    if k > cache_len {
        k = k % cache_len;
    }
    let nums_shift = cache[cache_len - k..].to_vec();
    let mut nums_shift = nums_shift;
    nums_shift.extend_from_slice(&cache);
    let mut result: Vec<Vec<i32>> = vec![];
    let row_len = grid[0].len();
    let mut j = 0;
    while j < nums_shift.len() - k {
        result.push(nums_shift[j..j + row_len].to_vec());
        j += row_len;
    }
    result
}
